url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/io/Generic_StaticIO.java

```diff
@@ -0,0 +1,2156 @@
+        do {
+            try {
+                file = File.createTempFile(
+                        prefix + Long.toString(System.currentTimeMillis()),
+                        suffix,
+                        parentDirectory);
+                abstractFileCreated = true;
+            } catch (IOException e) {
+                System.err.print(e.getMessage());
+                e.printStackTrace(System.err);
+                System.exit(Generic_ErrorAndExceptionHandler.IOException);
+            }
+        } while (!abstractFileCreated);
+        file.deleteOnExit();
+        return createNewFile(parentDirectory, file.getName());
+    }
+
+    /**
+     * @return createNewFile(new File(System.getProperty("user.dir")));
+     */
+    public static File createNewFile() {
+        // return createNewFile(
+        // new File( System.getProperty( "java.io.tmpdir" ) ) );
+        // return createNewFile(
+        // new File( System.getProperty( "user.home" ) ) );
+        return createNewFile(
+                new File(System.getProperty("user.dir")));
+    }
+
+    /**
+     * @return a File created by:      <code>
+     * return createNewFile(
+     *         parentDirectory,
+     *         new String(""),
+     *         new String(""));
+     * </code>
+     * @param parentDirectory Default extension prefix and suffix nothing.
+     */
+    public static File createNewFile(File parentDirectory) {
+        return createNewFile(
+                parentDirectory, "", "");
+    }
+
+    /**
+     * @return a File.
+     * @param parentDirectory
+     * @param prefix
+     * @param suffix
+     */
+    public static File createNewFile(
+            File parentDirectory,
+            String prefix,
+            String suffix) {
+        File file;
+        do {
+            file = new File(
+                    parentDirectory,
+                    prefix + Long.toString(System.currentTimeMillis()) + suffix);
+        } while (file.exists());
+        try {
+            if ((prefix + suffix).equalsIgnoreCase("")) {
+                file.mkdir();
+            } else {
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            System.err.print(e.getMessage());
+            e.printStackTrace(System.err);
+            System.exit(Generic_ErrorAndExceptionHandler.IOException);
+        }
+        return file;
+    }
+
+    /**
+     * Returns a newly created File which is a _InputFile if the filename. or a
+     * directory.
+     *
+     * @param parentDirectory
+     * @param filename
+     * @return
+     */
+    public static File createNewFile(
+            File parentDirectory,
+            String filename) {
+        File file = new File(parentDirectory, filename);
+        try {
+            if (filename.length() > 4) {
+                if (filename.charAt(filename.length() - 4) != '.') {
+                    file.mkdir();
+                } else {
+                    file.createNewFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`File.createNewFile` returns `false` if the named file already exists and an empty file is not created. Check the value returned by `createNewFile()`. Handle when the method returns `false`. Otherwise, you might silently overwrite an existing file which could lead to data loss. See the following diffs on GitHub that make this code change: [diff1](https://github.com/lingochamp/FileDownloader/commit/65069c18a4b1b4cfa758173ef49d69c6897fc77a#diff-2b6ad96df9bf9828c22f961e9e9a5b933ab2df7b2fb0f12d62c044309e1e2842L331), [diff2](https://github.com/m-szalik/javamail/commit/62b1cb45e32a998efbe6cfa09b06752869f590ae#diff-440ef5498fc2c802274cf136e7bd7d682bbc10c5976afb5af718984e24deced4L45)

Learn more: [File.createNewFile in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createNewFile--)